src/dev/ci_setup/setup.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -e
+
+
+###
+### determine directory of kibana project
+###
+dir="$(pwd)"
+if [ -f "$dir/package.json" ] && [ -f "$dir/.node-version" ]; then
+  echo "Setting up node.js and npm in $dir"
+else
+  echo "src/dev/ci_setup/setup.sh must be run within a kibana repo"
+  exit 1
+fi
+
+
+###
+### download node
+###
+nodeVersion="$(cat $dir/.node-version)"
+nodeUrl="https://nodejs.org/download/release/v$nodeVersion/node-v$nodeVersion-linux-x64.tar.gz"
+nodeDir="${NODE_DIR:-"$dir/.node"}/$nodeVersion"
+echo " -- node: version=v${nodeVersion} dir=$nodeDir"
+
+echo " -- setting up node.js"
+if [ -x "$nodeDir/bin/node" ] && [ "$($nodeDir/bin/node --version)" == "v$nodeVersion" ]; then
+  echo " -- reusing node.js install"
+else
+  if [ -d "$nodeDir" ]; then
+    echo " -- clearing invalid node.js install"
+    rm -rf "$nodeDir"
+  fi
+
+  echo " -- downloading node.js from $nodeUrl"
```

Comment:
thoughts on doing a checksum here?  if it's making it all the way to releases IMO it's worth it, even if it's manual updates

Reply:
The node used here won't be used for releases, the build process downloads it's own version of node for each platform and verifies the checksums there. We're downloading over https, so I don't think it's worth the complexity. Maybe it's not that bad, but that's what I'm thinking